frontend/src/components/Login.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import TextField from '@material-ui/core/TextField'
+import IconButton from '@material-ui/core/IconButton'
+import Input from '@material-ui/core/Input'
+import InputLabel from '@material-ui/core/InputLabel'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import Visibility from '@material-ui/icons/Visibility'
+import VisibilityOff from '@material-ui/icons/VisibilityOff'
+
+import { login, signUp } from '../reducers/user'
+import { CustomButton } from './CustomButton'
+import { Profile } from './Profile'
+import {
+  Container,
+  Wrapper,
+  Form,
+  ButtonWrapper,
+  Title,
+  SubTitle
+} from '../styles/Style'
+
+export const Login = () => {
+  const [name, setName] = useState("")
+  const [email, setEmail] = useState("")
+  const [password, setPassword] = useState("")
+  const [showPassword, setShowPassword] = useState(false)
+
+  const errorMessage = useSelector((store) => store.user.login.errorMessage)
+  const accessToken = useSelector((store) => store.user.login.accessToken)
+
+  const dispatch = useDispatch()
+
+  // Form validation
+  const minimumNameLength = { minLength: 2, maxLength: 20 }
+  const minimumPasswordLength = { minLength: 5 }
+  const validEmail = { pattern: "^[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,}$" }
+  //cannot start with @ and has to include the pattern of xxxx@xxx.se
+
+  // Toggle password visibility
+  const handleClickShowPassword = () => setShowPassword(!showPassword)
+
+  const handleLogin = (event) => {
+    event.preventDefault()
+    dispatch(login(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  const handleSignUp = (event) => {
+    event.preventDefault()
+    dispatch(signUp(name, email, password))
+    setName('')
+    setEmail('')
+    setPassword('')
+  }
+
+  return (
+    <Container>
+      <Wrapper>
+        {!accessToken ? (
+          <Form onSubmit={(event) => event.preventDefault()}>
```

Comment:
It might be a good idea to have two separate forms for login and signup. One reason is that the user on login then don't have to enter their name. And also (but we are a bit unsure about this) it might not not be best practice to have two submit-buttons on the same form.